pkg/apis/acme/v1alpha2/types_issuer.go

```diff
@@ -204,6 +209,23 @@ type ACMEChallengeSolverHTTP01IngressPodSpec struct {
 	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
 }
 
+type ACMEChallengeSolverHTTP01IngressTemplate struct {
+	// ObjectMeta overrides for the ingress used to solve HTTP01 challenges.
+	// Only the 'labels' and 'annotations' fields may be set.
+	// If labels or annotations overlap with in-built values, the values here
+	// will override the in-built values.
+	// +optional
+	ACMEChallengeSolverHTTP01IngressObjectMeta `json:"metadata,omitempty"`
```

Comment:
I think we should remove `omitempty` here as it will have no effect because this field is not a pointer. Keep `// +optional` however as this is what dictates what our OpenAPI schema says 😄

Reply:
For the template and meta I mostly copied the `ACMEChallengeSolverHTTP01IngressPodTemplate` and `ACMEChallengeSolverHTTP01IngressPodObjectMeta` just above. Should I add those changes there as well while I'm at it?

Reply:
Yes please! 😄